posthog-inspector-docs (#435)
avohq/docs: pages/workspace/connect-inspector-to-posthog.mdx

---
layout: 'guide'
title: 'Send data from PostHog to Avo Inspector'
---

import Link from '../../components/Link';

## Use Avo Inspector as a PostHog destination

If you are already sending your data to PostHog you can stream it to Avo Inspector without adding any code to your codebase.

> PostHog has an official plugin maintained by PostHog that sends data directly to the Avo Inspector. If you are using Posthog you can set it up within a few minutes.

Reviewer: ```suggestion
PostHog has an official plugin maintained by PostHog that sends data directly to the Avo Inspector. If you are using PostHog you can set it up within a few minutes.
```

> The Plugin is a side effect of the Avo Inspector analyzing your events and keeping your PostHog events stream unchanged.

After connecting the PostHog plugin to a source the event schemas of all events seen in PostHog will automatically be sent to Avo Inspector and you'll be able to monitor the quality of your tracking implementation without any changes in your code!

> > Remember that production data takes up to 2 hours to appear in the Inspector dashboard, while development data is available without any delay

Reviewer: Should we also highlight that Inspector only receives the event schemas, and not the actual data?

## Guide to connect PostHog and Avo Inspector
#### Step 1. Set up your Avo source

An Avo Source is the first thing you need to connect your PostHog data to Avo Inspector.

Open `Sources` in the Avo sidebar or `Manage Sources` in the Inspector tab

<center style={{marginBottom: 16 + "px"}}>
  <img
    width="100%"
    src={require('../../images/workspace/nav-inspector-sources-highlighted.png')}
    alt="Navigation to the Inspector sources screen in Avo"
  />
</center>

> If you already have the source, pick it. If you don’t have the source corresponding to your app, <Link passHref href="/workspace/inspector#configuring-sources"><a>add a new one</a></Link>.

Reviewer: ```suggestion
If you already have the source, open it in the dashboard. If you don’t have the source corresponding to your app, <Link passHref href="/workspace/inspector#configuring-sources"><a>add a new one</a></Link>.
```

<center style={{marginBottom: 16 + "px"}}>
  <img
    src={require('../../images/add-source.png')}
    alt="Image of the Add Source button in Avo"
  />
</center>

> In the source details, go to "Inspector Setup" and there you’ll see options to integrate Inspector. Pick PostHog to see the integration code snippets.

Reviewer: ```suggestion
In the source details, go to "Inspector Setup" and there you’ll see options to integrate Inspector. Pick PostHog to see the integration guide.
```

<center style={{marginBottom: 16 + "px"}}>
  <img
    height="400"
    src={require('../../images/guide-posthog-to-inspector/source-setup.png')}
    alt="Source settings in Avo"
  />
</center>

Copy the API key, you will need to use it in PostHog for the next step.

#### Step 2. Enable Avo Inspector plugin in PostHog

- Open PostHog and log in into your workspace

- Pick `Plugins` in the sidebar 

- Search for `Avo Inspector` in the search box

<center style={{marginBottom: 16 + "px"}}>
  <img
    src={require('../../images/guide-posthog-to-inspector/avo-plugin.png')}
    alt="PostHog navigation to the Plugin menu"
  />
</center>

Enable the plugin and paste the API key

Save it by pressing `Save` button under the code editor.

<center style={{marginBottom: 16 + "px"}}>
  <img
    src={require('../../images/guide-posthog-to-inspector/insert-api-key.png')}
    alt="PostHog insert avo API key"
  />
</center>

#### Step 3. Test your PostHog plugin integration on Dev.

> > You can test the production PostHog Plugin in the same way, but keep in mind that the production data shows with a ~2 hours delay in the Inspector dashboard, while development data is shown without delay

Reviewer: ```suggestion
> You can test the production PostHog Plugin in the same way, but keep in mind that the production data shows with a ~2 hours delay in the Inspector dashboard, while development and staging data is shown without delay
```

Monitor the logs for the Avo Inspector plugin in PostHog and verify events are being sent to the Avo Inspector.

<center style={{marginBottom: 16 + "px"}}>
  <img
    src={require('../../images/guide-posthog-to-inspector/logs.png')}
    alt="Posthog plugin logs"
  />
</center>

Then navigate to Avo, pick `Inspector` in the sidebar and pick Development environment

<center style={{marginBottom: 16 + "px"}}>
  <img
    src={require('../../images/guide-segment-to-inspector/inspector-development-env.png')}
    alt="Environment picker in Inspector"
  />
</center>

You’ll see the event in your dashboard under the source from where you got the snippet from.

<center style={{marginBottom: 16 + "px"}}>
  <img
    height="500"
    src={require('../../images/guide-segment-to-inspector/source-in-inspector.png')}
    alt="Source overview in Inspector"
  />
</center>

> This means that the PostHog plugin is working.

Reviewer: ```suggestion
This means that the PostHog plugin is working 🎉
```
